crates/rules-uniffi/src/builder.rs

```diff
@@ -0,0 +1,279 @@
+#![allow(clippy::new_without_default)]
+#![allow(dead_code)]
+#![allow(unused_variables)]
+
+use std::sync::{Arc, RwLock};
+
+use sargon::IndexSet;
+
+use crate::prelude::*;
+
+#[derive(Debug, uniffi::Object)]
+pub struct SecurityShieldBuilder {
+    wrapped: RwLock<Option<MatrixBuilder>>,
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, uniffi::Object)]
+#[uniffi::export(Debug, Eq, Hash)]
+pub struct SecurityStructureOfFactorSourceIds {
+    wrapped_matrix: MatrixWithFactorSourceIds,
+    name: String,
```

Comment:
this will have an ID as well... the models here in this crate is not 100% accurate yet. What I want reviewed is probably just the implementation of the wrapping? And the API